SwiftWisdomTests/StandardLibrary/String/String+IndexingTests.swift

```diff
@@ -2,6 +2,30 @@ import XCTest
 
 class String_IndexingTests: XCTestCase {
     
+    func testRangeSubscriptReturnsProperString() {
+        XCTAssertEqual("hello"[0..<0], "")
```

Comment:
While not in this PR, I would love if we can take on @bobgilmore's concern and just try seeing if the indexing works on these two(bob, feel free to chime in if you have specific examples in mind)

1. la ingeniería (assert that [0...12] returns full string)
2. 工程师 (assert that [0...3] returns full string)

If those don't work, it's worth to document that as gotchas. @Dennis-Cherchenko if you don't end up doing this can you add a github issue?